### ISSUE TITLE
Bug: Loading compressed data fails silently (no error message, the application simply hangs up)

### ISSUE THREAD

**Issue by AugustDev** [Lightning-AI/litdata#341]
## 🐛 Bug

I am trying to install and use litdata in clean environment, but there are few problems
1. Loading compressed data fails silently (no error message, the application simply hangs up)
2. Importing litdata gives a warning about numpy 1.x
3. Loading the data gives `RuntimeError: Could not infer dtype of numpy.int64`

```
A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.2 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.

Traceback (most recent call last):  File "/Users/wpc/Projects/cfdx/data/main.py", line 1, in <module>
    from litdata import StreamingDataset, StreamingDataLoader, optimize
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/litdata/__init__.py", line 16, in <module>
    from litdata.processing.functions import map, merge_datasets, optimize, walk
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/litdata/processing/functions.py", line 28, in <module>
    import torch
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/__init__.py", line 1477, in <module>
    from .functional import *  # noqa: F403
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/functional.py", line 9, in <module>
    import torch.nn.functional as F
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/nn/__init__.py", line 1, in <module>
    from .modules import *  # noqa: F403
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/nn/modules/__init__.py", line 35, in <module>
    from .transformer import TransformerEncoder, TransformerDecoder, \
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/nn/modules/transformer.py", line 20, in <module>
    device: torch.device = torch.device(torch._C._get_default_device()),  # torch.device('cpu'),
/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/nn/modules/transformer.py:20: UserWarning: Failed to initialize NumPy: _ARRAY_API not found (Triggered internally at /Users/runner/work/pytorch/pytorch/pytorch/torch/csrc/utils/tensor_numpy.cpp:84.)
  device: torch.device = torch.device(torch._C._get_default_device()),  # torch.device('cpu'),
Dataset length: 1000
0
Traceback (most recent call last):
  File "/Users/wpc/Projects/cfdx/data/main.py", line 29, in <module>
    main()
  File "/Users/wpc/Projects/cfdx/data/main.py", line 23, in main
    for batch in data_loader:
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/litdata/streaming/dataloader.py", line 624, in __iter__
    for batch in super().__iter__():
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 631, in __next__
    data = self._next_data()
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 675, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 42, in fetch
    return self.collate_fn(data)
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/utils/data/_utils/collate.py", line 277, in default_collate
    return collate(batch, collate_fn_map=default_collate_fn_map)
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/utils/data/_utils/collate.py", line 125, in collate
    return collate_fn_map[collate_type](batch, collate_fn_map=collate_fn_map)
  File "/Users/wpc/Projects/cfdx/data/venv3/lib/python3.9/site-packages/torch/utils/data/_utils/collate.py", line 187, in collate_numpy_scalar_fn
    return torch.as_tensor(batch)
RuntimeError: Could not infer dtype of numpy.int64
```

### To Reproduce

Create `main.py` with 

```py
from litdata import StreamingDataset, StreamingDataLoader, optimize


def item(x):
    return x


def create_dataset():
    optimize(
        fn=item,
        inputs=list(range(1000)),
        output_dir="my_optimized_dataset",
        chunk_bytes="64MB",
    )


def main():
    dataset = StreamingDataset("./my_optimized_dataset")
    print(f"Dataset length: {len(dataset)}")
    print(dataset[0])

    data_loader = StreamingDataLoader(dataset)
    for batch in data_loader:
        print(batch)


if __name__ == "__main__":
    create_dataset()
    main()

```

Steps to reproduce

```
virtualenv venv
source venv/bin/activate
pip install 'litdata[extras]'
python main.py
```

### Expected behavior

1. If user is trying to load data that was saved with compression and has no compression library installed it should give an error. Ideally compression libraries are installed together with `pip install litdata`
2. There is no warning about numpy versions
3. Iterating through data loader prints the data and there is no `Could not infer dtype of numpy.int64` error

### Environment

- Python Version 3.9.13
- OS: MacOS 14.6.1
- pip

### Other

What fixes the errors is 

`pip install "numpy<2"`

However, I worry about using older versions of numpy.

**Comment by vineet1401** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by bhimrazy** [comment]
Hi @AugustDev, Thank you for bringing these issues to our attention. 
I tried to reproduce the bugs. I was able to replicate the first one, but I haven't had any luck with the second and third.

**Comment by AugustDev** [comment]
Hi @bhimrazy thanks for the reply. I am myself not able to reproduce the 2'nd 3rd ones on different machines. Not sure why. I think we can ignore 2nd and 3rd point for now.

**Comment by bhimrazy** [comment]
Hi @AugustDev, We've just released version v0.2.25 with the fix. 
Feel free to give it a try.

Thank you!
